projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts

```diff
@@ -75,8 +78,6 @@ export class MeasurementsPickerComponent implements OnInit, OnDestroy {
   };
 
   ngOnDestroy(): void {
-    if (this.subscriptions) {
-      this.subscriptions.unsubscribe();
-    }
+    this.subscriptionsTerminator$.next();
```

Comment:
It should be completed also

Reply:
@dvladir I did the changes, but it seems to work without takeUntil. are you sure it is needed? maybe it's safer to complete it, but takeUntil will react when a .next() is called on the observable. Do I miss something?

Reply:
@lucianbaciu 
I played around a bit. Yes you're right `takeUntil` completes the source stream, when subject's `next` is invoked.
So `complete` inactivation not required for termination - I was wrong about it

But anyway completion of any subjects, that were declared in the component - is a good practice. I highly recommend to keep it.